ui/app/pages/swaps/awaiting-swap/view-on-ether-scan-link/view-on-ether-scan-link.test.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import { renderWithProvider } from '../../../../../../test/jest';
+import ViewOnEtherScanLink from '.';
+
+const createProps = (customProps = {}) => {
+  return {
+    txHash:
+      '0x58e5a0fc7fbc849eddc100d44e86276168a8c7baaa5604e44ba6f5eb8ba1b7eb',
+    blockExplorerUrl: 'https://block.explorer',
+    isCustomBlockExplorerUrl: false,
+    ...customProps,
+  };
+};
+
+describe('ViewOnEtherScanLink', () => {
+  it('renders the component with initial props', () => {
+    const { container, getByText } = renderWithProvider(
+      <ViewOnEtherScanLink {...createProps()} />,
+    );
+    expect(getByText('View on Etherscan')).toBeInTheDocument();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders the component with a custom block explorer link', () => {
+    const { container, getByText } = renderWithProvider(
+      <ViewOnEtherScanLink
+        {...createProps({
+          blockExplorerUrl: 'https://custom-blockchain.explorer',
+          isCustomBlockExplorerUrl: true,
+        })}
+      />,
+    );
+    expect(getByText('View at custom-blockchain.explorer')).toBeInTheDocument();
```

Comment:
Is it possible for us to test what the URL will go to here instead of just checking the link text? It'd help to ensure that changes made to the URL but not the text don't pass through our test cases.

Reply:
Great point, will add an assertion for a URL.

Reply:
I've checked the code and we don't render a link, but open a new tab via an `onClick` event on a `div` element:
```onClick={() => global.platform.openTab({ url: blockExplorerUrl })}```

I will be testing events in following PRs, so I would prefer to do it there.